src/graph/graph_hub.c

```diff
@@ -143,7 +143,7 @@ void CreateEdge
 
 	// add edge creation operation to undo log
 	if(log == true) {
-		UndoLog *undo_log = QueryCtx_GetUndoLog();
+		UndoLog undo_log = *QueryCtx_GetUndoLog();
```

Comment:
Maybe `QueryCtx_GetUndoLog` should return `UndoLog` instead of `UndoLog*` ?

Reply:
we need the pointer so we can null it in the rollback

Reply:
So maybe we should change this to:
`QueryCtx_Rollback` which will take care of freeing / nullifying the UndoLog ?